### PR TITLE
fixup examples based on default pin #define-s and also support them being undefined

### PR DESCRIPTION
**Pull request by kilograham**

### CODE REVIEW

File: i2c/bus_scan/bus_scan.c [hunk 1]
```diff
@@ -33,13 +34,18 @@ bool reserved_addr(uint8_t addr) {
...
     stdio_init_all();
-
-    // This example will use I2C0 on GPIO4 (SDA) and GPIO5 (SCL)
-    i2c_init(i2c0, 100 * 1000);
-    gpio_set_function(4, GPIO_FUNC_I2C);
-    gpio_set_function(5, GPIO_FUNC_I2C);
-    gpio_pull_up(4);
-    gpio_pull_up(5);
+#if !defined(i2c_default) || !defined(PICO_DEFAULT_I2C_SDA_PIN) || !defined(PICO_DEFAULT_I2C_SCL_PIN)
+#warning i2c/bus_scane example requires a board with I2C pins
```
Comment (lurch) [inline i2c/bus_scan/bus_scan.c:38]: bus_scane -> bus_scan

File: i2c/lcd_1602_i2c/lcd_1602_i2c.c [hunk 2]
```diff
@@ -129,14 +130,17 @@ void lcd_init() {
...
 
 int main() {
-    // This example will use I2C0 on GPIO4 (SDA) and GPIO5 (SCL)
-    i2c_init(I2C_PORT, 100 * 1000);
-    gpio_set_function(4, GPIO_FUNC_I2C);
-    gpio_set_function(5, GPIO_FUNC_I2C);
-    gpio_pull_up(4);
-    gpio_pull_up(5);
+#if !defined(i2c_default) || !defined(PICO_DEFAULT_I2C_SDA_PIN) || !defined(PICO_DEFAULT_I2C_SCL_PIN)
+    #warning i2c/bus_scane example requires a board with I2C pins
```
Comment (lurch) [inline i2c/lcd_1602_i2c/lcd_1602_i2c.c:134]: bus_scane -> lcd_1602_i2c

File: i2c/mpu6050_i2c/mpu6050_i2c.c [hunk 3]
```diff
@@ -70,23 +70,29 @@ static void mpu6050_read_raw(int16_t accel[3], int16_t gyro[3], int16_t *temp) {
...
 
     *temp = buffer[0] << 8 | buffer[1];
 }
+#endif
 
 int main() {
     stdio_init_all();
-
+#if !defined(i2c_default) || !defined(PICO_DEFAULT_I2C_SDA_PIN) || !defined(PICO_DEFAULT_I2C_SCL_PIN)
+    #warning i2c/bus_scane example requires a board with I2C pins
```
Comment (lurch) [inline i2c/mpu6050_i2c/mpu6050_i2c.c:83]: bus_scane -> mpu6050_i2c

File: i2c/bus_scan/bus_scan.c [hunk 4]
```diff
@@ -33,13 +34,18 @@ bool reserved_addr(uint8_t addr) {
...
+    puts("Default I2C pins were not defined");
```
Comment (lurch) [inline i2c/bus_scan/bus_scan.c:39]: Just out of curiosity - why the usage of `puts` here, when `printf` is used everywhere else?

File: picoboard/button/button.c [hunk 5]
```diff
@@ -53,10 +53,14 @@ bool __no_inline_not_in_flash_func(get_bootsel_button)() {
...
 
 int main() {
+#ifndef PICO_DEFAULT_LED_PIN
+#warning picobooard/button example requires a board with a regular LED
+#else
     gpio_init(PICO_DEFAULT_LED_PIN);
     gpio_set_dir(PICO_DEFAULT_LED_PIN, GPIO_OUT);
     while (true) {
-        gpio_put(PICO_DEFAULT_LED_PIN, get_bootsel_button());
+        gpio_put(PICO_DEFAULT_LED_PIN, get_bootsel_button() ^ PICO_DEFAULT_LED_PIN_INVERTED);
```
Comment (lurch) [inline picoboard/button/button.c:62]: Does this `PICO_DEFAULT_LED_PIN_INVERTED` also need to be used in all the other examples that use `PICO_DEFAULT_LED_PIN` ?

File: i2c/bus_scan/bus_scan.c [hunk 6]
```diff
@@ -33,13 +34,18 @@ bool reserved_addr(uint8_t addr) {
...
+#else
+    // This example will use I2C0 on the default SDA and SCL pins (4, 5 on a Pico)
+    i2c_init(i2c_default, 100 * 1000);
+    gpio_set_function(PICO_DEFAULT_I2C_SDA_PIN, GPIO_FUNC_I2C);
+    gpio_set_function(PICO_DEFAULT_I2C_SCL_PIN, GPIO_FUNC_I2C);
+    gpio_pull_up(PICO_DEFAULT_I2C_SDA_PIN);
+    gpio_pull_up(PICO_DEFAULT_I2C_SCL_PIN);
+    // Make the I2C pins available to picotool
+    bi_decl(bi_2pins_with_func(PICO_DEFAULT_I2C_SDA_PIN, PICO_DEFAULT_I2C_SCL_PIN, GPIO_FUNC_I2C));
```
Comment (lurch) [inline i2c/bus_scan/bus_scan.c:48]: Would this PR be a good place to add `bi_decl` stuff to all the other examples too, or would that make more sense in a separate PR?

File: i2c/bus_scan/bus_scan.c (hunk 4 above)
Comment (kilograham) [reply i2c/bus_scan/bus_scan.c:39]: moon phase

File: picoboard/button/button.c (hunk 5 above)
Comment (kilograham) [reply picoboard/button/button.c:62]: i think most of the reset just flash it, so it doesn't much matter which way (at least that was my thinking)

File: i2c/bus_scan/bus_scan.c (hunk 6 above)
Comment (kilograham) [reply i2c/bus_scan/bus_scan.c:48]: separate PR - feel free :-)

File: picoboard/button/button.c (hunk 5 above)
Comment (lurch) [reply picoboard/button/button.c:62]: Good point. (assuming that the 'on' and 'off' periods are the same length, which they probably are!)

File: i2c/bus_scan/bus_scan.c (hunk 6 above)
Comment (lurch) [reply i2c/bus_scan/bus_scan.c:48]: LOL, I'm not that familiar with the binary_info stuff yet... maybe towards the end of the week :wink:

**Comment by liamfraser** [comment]
LGTM apart from the couple of typos @lurch pointed out.

File: i2c/mpu6050_i2c/mpu6050_i2c.c [hunk 7]
```diff
@@ -70,23 +70,29 @@ static void mpu6050_read_raw(int16_t accel[3], int16_t gyro[3], int16_t *temp) {
...
 
     *temp = buffer[0] << 8 | buffer[1];
 }
+#endif
 
 int main() {
     stdio_init_all();
-
+#if !defined(i2c_default) || !defined(PICO_DEFAULT_I2C_SDA_PIN) || !defined(PICO_DEFAULT_I2C_SCL_PIN)
+    #warning i2c/bus_scan example requires a board with I2C pins
```
Comment (liamfraser) [inline i2c/mpu6050_i2c/mpu6050_i2c.c:83]: The example here isn't i2c/bus_scan. Perhaps have a macro for the example name? Not big deal so happy to approve anyway.

Comment (lurch) [reply i2c/mpu6050_i2c/mpu6050_i2c.c:83]: I can submit a followup fix PR

**Review by liamfraser — APPROVED** [review]

File: i2c/mpu6050_i2c/mpu6050_i2c.c (hunk 7 above)
Comment (lurch) [reply i2c/mpu6050_i2c/mpu6050_i2c.c:83]: Fixed in #75
